File: backend/app_api/domain/values/timestamp.py
from dataclasses import dataclass, field
from datetime import datetime
from typing import Optional

@dataclass(kw_only=True)
class TimestampModel:
    created_at: datetime = field(default_factory=datetime.now)
    updated_at: datetime = field(default_factory=datetime.now)
    deleted_at: Optional[datetime] = None

File: frontend/src/domain/values/Timestamp.ts
export class TimeStamp {
    created_at: Date | null;
    updated_at: Date | null;
    deleted_at: Date | null;

    constructor(created_at: Date | null, updated_at: Date | null, deleted_at: Date | null) {
        this.created_at = created_at;
        this.updated_at = updated_at;
        this.deleted_at = deleted_at;
    }
}